wemake_python_styleguide/visitors/ast/imports.py

```diff
@@ -128,6 +133,101 @@ def _check_protected_import_from_members(
                 self._error_callback(ProtectedModuleMemberViolation(node))
 
 
+@final
```

Comment:
Let's move this out of visitors into logic.

Reply:
whole class? 
This will break import linter `layers` contract, because of importing `violations` in `logic`

Reply:
Why? This class does not look like it has something to do with `violations`:

```python
@final
class _ImportedObjectInfo(NamedTuple):
    node: AnyImport
    # `position` is needed to distinguish imported names in single
    # import statement (`import x, y, z`)
    position: int
    name: str
```

Reply:
ahh, sorry, I was thinking about `_ImportConflictValidator` instead of `_ImportedObjectInfo`.
It will be moved in next commit